### PR TITLE
Improve urbdrc device registering via /usb:id and addr

### PR DESCRIPTION
**Pull request by lukasfink1**
This pr adds 3 improvements to the urbdrc /usb:id, addr  and dev  command line options:

1. It fixes them. Currently they don’t work, because the devices are attempted to be added before the urbdrc addin is initialized, which fails silently. A function is added to the `IUDEVMAN` struct, that gets called after initialization. Parsing of the device option is done there. (513cfc1)
2. It adds hotplug to the `/usb:id,dev` option. Devices, that are connected at runtime, are added when they match a specified vid+pid-pair. (5724ffc)
3. It adds the ability to specify devices via vid+pid-pairs and bus+addr-pairs simultaneously. This is done by using the id and addr option directly: `/usb:id:0123:4567#89ab:cdef,addr:01:02#03:04`. The dev option can still be used like it could before. (0155a5b)

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/urbdrc/client/libusb/libusb_udevman.c [hunk 1]
```diff
@@ -542,18 +556,27 @@ static int hotplug_callback(struct libusb_context* ctx, struct libusb_device* de
...
 
-	if (device_is_filtered(dev, &desc, event))
-		return 0;
-
 	switch (event)
 	{
 		case LIBUSB_HOTPLUG_EVENT_DEVICE_ARRIVED:
-			if (idevman->isAutoAdd(idevman))
-				add_device(idevman, DEVICE_ADD_FLAG_ALL, bus, addr, desc.idVendor, desc.idProduct);
+			for (size_t i = 0; i < udevman->num_hotplug_vid_pids; i++)
```
Comment (akallabeth) [inline channels/urbdrc/client/libusb/libusb_udevman.c:562]: Variable declaration mid code breaks older compilers

File: channels/urbdrc/client/libusb/libusb_udevman.c [hunk 2]
```diff
@@ -65,6 +75,10 @@ struct _UDEVMAN
 	IUDEVICE* head; /* head device in linked list */
 	IUDEVICE* tail; /* tail device in linked list */
 
+	LPSTR devices_vid_pid;
+	LPSTR devices_addr;
+	VID_PID_PAIR hotplug_vid_pids[MAX_HOTPLUG_VID_PIDS];
```
Comment (akallabeth) [inline channels/urbdrc/client/libusb/libusb_udevman.c:80]: Maybe use an `wArrayList` or similar as dynamic structure for this

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by lukasfink1** [comment]
I also fixed a bug found by scan-build where a null function pointer gets invoked when the udevman initialization fails early.

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/urbdrc/client/libusb/libusb_udevman.c [hunk 3]
```diff
@@ -418,13 +433,15 @@ static void udevman_free(IUDEVMAN* idevman)
 		return;
 
 	udevman->running = FALSE;
-	WaitForSingleObject(udevman->thread, INFINITE);
+	if (udevman->thread)
+		WaitForSingleObject(udevman->thread, INFINITE);
 
 	udevman_unregister_all_udevices(idevman);
 	CloseHandle(udevman->devman_loading);
 	CloseHandle(udevman->thread);
```
Comment (akallabeth) [inline channels/urbdrc/client/libusb/libusb_udevman.c:441]: This should be covered by the `NULL` check too.

File: channels/urbdrc/client/libusb/libusb_udevman.c [hunk 4]
```diff
@@ -542,18 +560,21 @@ static int hotplug_callback(struct libusb_context* ctx, struct libusb_device* de
...
 	{
 		case LIBUSB_HOTPLUG_EVENT_DEVICE_ARRIVED:
-			if (idevman->isAutoAdd(idevman))
-				add_device(idevman, DEVICE_ADD_FLAG_ALL, bus, addr, desc.idVendor, desc.idProduct);
+			pair.vid = desc.idVendor;
+			pair.pid = desc.idProduct;
+			if (ArrayList_Contains(udevman->hotplug_vid_pids, &pair) ||
+			    (udevman->iface.isAutoAdd(&udevman->iface) &&
+			     !device_is_filtered(dev, &desc, event)))
+				add_device(&udevman->iface, DEVICE_ADD_FLAG_ALL, bus, addr, desc.idVendor,
```
Comment (akallabeth) [inline channels/urbdrc/client/libusb/libusb_udevman.c:571]: Braces could improve readability here

Comment (lukasfink1) [reply channels/urbdrc/client/libusb/libusb_udevman.c:571]: But CloseHandle already checks for null pointers itself

File: channels/urbdrc/client/libusb/libusb_udevman.c (hunk 3 above)
Comment (lukasfink1) [reply channels/urbdrc/client/libusb/libusb_udevman.c:441]: But CloseHandle already checks for null pointers itself

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/urbdrc/client/libusb/libusb_udevman.c (hunk 3 above)
Comment (akallabeth) [reply channels/urbdrc/client/libusb/libusb_udevman.c:441]: @lukasfink1 ours does, windows is different. (depends on version)

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**